libsel4vm/include/sel4vm/guest_vm_util.h

```diff
@@ -72,6 +72,30 @@ static inline vm_vcpu_t *vm_find_free_unassigned_vcpu(vm_t *vm)
     return NULL;
 }
 
+/***
+ * @function vm_find_free_unassigned_cpu(vm)
+ * Find an unallocated physical cpu
+ * @param {vm_t *} vm           A handle to the vm owning the vcpu
+ * @return                      -1 if no pcpu can be found, otherwise the available pcpu
+ */
+static inline int vm_find_free_unassigned_cpu(vm_t *vm)
+{
+    for (int core = 0; core < vm->num_vcpus; core++) {
+        int core_allocated = 0;
+        for (int i = 0; i < vm->num_vcpus; i++) {
+            if (vm->vcpus[i]->target_cpu == core) {
+                core_allocated = 1;
+                break;
+            }
+        }
+        if (core_allocated) {
```

Comment:
reverse the ceck, then there is no need for the explicit `continue` and the algorithm is a bit easier to read
```suggestion
        if (!core_allocated) { 
            return core
        }
```

Reply:
Done